src/app/editor/shared/services/schema-validation.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class SchemaValidationService {
+  
+  private static formatRegexps = {
+    date: /^\d\d\d\d-[0-1]\d-[0-3]\d$/,
+    email: /^[a-z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?(?:\.[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?)*$/i,
+    isbn: /^ISBN\s(?=[-0-9xX ]{13}$)(?:[0-9]+[- ]){3}[0-9]*[xX0-9]$/,
```

Comment:
I actually didn't know about `pattern`. I thought `format` can have either a regexp pattern or a regexp pattern keyword like `email`, `url` etc. I should consider `pattern` for the implementation.

Also, if `pattern` exists then we shouldn't have regexp pattern in `format` right?
like here: https://github.com/inspirehep/record-editor/blob/master/src/assets/hep.json#L879 @kaplun 

No, it was really easy to implement it this way and it is more efficient than 3rd party which I suppose only support validation of entire record. But in this way we only validate single changed value, since we already track changes.